### PR TITLE
Fix regression with duplicate `#[macro_export] macro_rules!`

### LINKED ISSUE #38715: Regression in test suite of probor-0.2.1 in Rust 1.15
https://github.com/tailhook/probor at d0be60ec37ded2eee0f851887e2d7c06fe2e7021

```
101 brian@ip-10-145-43-250:~/dev/probor/rust⟫ rustc +beta -Vv
rustc 1.15.0-beta.1 (d9a0f0df7 2016-12-19)
binary: rustc
commit-hash: d9a0f0df7051c603011d6b60fbdd155318fc47f3
commit-date: 2016-12-19
host: x86_64-unknown-linux-gnu
release: 1.15.0-beta.1
LLVM version: 3.9
```

```
brian@ip-10-145-43-250:~/dev/probor/rust⟫ cargo +beta test
   Compiling byteorder v1.0.0
   Compiling libc v0.2.18
   Compiling byteorder v0.5.3
   Compiling cbor-codec v0.5.0
   Compiling probor v0.2.1 (file:///mnt2/dev/probor/rust)
error[E0428]: a macro named `_probor_pattern` has already been defined in this module

error: aborting due to previous error

Build failed, waiting for other jobs to finish...
error: Could not compile `probor`.

```

cc @jseyfried I haven't seen one quite like this.

cc @tailhook

### PR DESCRIPTION
**Pull request by jseyfried**
Fixes #38715.
r? @nrc

### CODE REVIEW

**Comment by brson** [comment]
Thanks @jseyfried !

File: src/test/run-pass/auxiliary/issue_38715.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
```
Comment (nrc) [inline src/test/run-pass/auxiliary/issue_38715.rs]: Could you add a comment explaining what is being tested please?

**Review by nrc — APPROVED** [review]
r = me with the comment in the test

File: src/test/run-pass/auxiliary/issue_38715.rs (hunk 1 above)
Comment (jseyfried) [inline src/test/run-pass/auxiliary/issue_38715.rs]: Done.

**Comment by nikomatsakis** [comment]
Small patch. Low impact. Accepted for beta. cc @rust-lang/compiler

**Comment by alexcrichton** [comment]
@bors: p=1

(this is approved, we'll want to backport)

**Comment by alexcrichton** [comment]
@bors: retry

* osx linker segfaulted

**Comment by alexcrichton** [comment]
@bors: retry

* sccache wedged?

**Comment by alexcrichton** [comment]
@bors: retry

* network failure

**Comment by alexcrichton** [comment]
@bors: retry

* network failure

**Comment by alexcrichton** [comment]
@bors: retry

* osx linker segfaulted

**Comment by alexcrichton** [comment]
@bors: retry

* network error

**Comment by alexcrichton** [comment]
@bors: retry

* osx linker segfaulted

**Pull request merged**